tests/logger_tests.cpp

```diff
@@ -47,28 +47,83 @@ class raii_restore_env {
   bool is_set_{false};
 };
 
-TEST(Adaptor, first)
+/**
+ * @brief Verifies the specified log file contains the expected events.
+ *
+ * Events in the log file are expected to occur in the same order as in `expected_events`.
+ *
+ * @note: This function accounts for the fact that `device_memory_resource` automatically pads
+ * allocations to a multiple of 8 bytes by rounding up the expected allocation sizes to a multiple
+ * of 8.
+ *
+ * @param filename Name of CSV log file generated from `logging_resource_adaptor`
+ * @param expected_events List of expected (de)allocation events
+ */
+void expect_log_events(std::string const& filename,
+                       std::vector<rmm::detail::event> const& expected_events)
 {
+  auto actual_events = rmm::detail::parse_csv(filename);
+
+  std::equal(expected_events.begin(),
+             expected_events.end(),
+             actual_events.begin(),
+             [](auto expected, auto actual) {
+               EXPECT_EQ(expected.act, actual.act);
+               // device_memory_resource automatically pads an allocation to a multiple of 8 bytes
+               EXPECT_EQ(rmm::detail::align_up(expected.size, 8), actual.size);
+               EXPECT_EQ(expected.pointer, actual.pointer);
+               return true;
+             });
+}
+
+TEST(Adaptor, FilenameConstructor)
+{
+  std::string filename{"logs/test1.txt"};
   rmm::mr::cuda_memory_resource upstream;
+  rmm::mr::logging_resource_adaptor<rmm::mr::cuda_memory_resource> log_mr{&upstream, filename};
 
-  rmm::mr::logging_resource_adaptor<rmm::mr::cuda_memory_resource> log_mr{&upstream,
-                                                                          "logs/test1.txt"};
+  auto p0 = log_mr.allocate(100);
+  auto p1 = log_mr.allocate(42);
+  log_mr.deallocate(p0, 100);
+  log_mr.deallocate(p1, 42);
+  log_mr.flush();
 
-  auto p = log_mr.allocate(100);
-  log_mr.deallocate(p, 100);
+  using rmm::detail::action;
+  using rmm::detail::event;
+
+  std::vector<event> expected_events{{action::ALLOCATE, 100, p0},
```

Comment:
No thread ID or stream ID in the tests?

Reply:
So testing the thread id is tricky. There's not really a guarantee that that thread id `spdlog` is capturing is the same thing as what's returned by `std::thread::get_id()` (it almost certainly is the same, but it's not guaranteed). 

What's more problematic is round tripping a `thread_id`. `std::thread::id` isn't at all guaranteed to be a numeric value. So whatever value is written by `spdlog` for the thread id isn't comparable to `std::thread::id` without an unsafe conversion.

Reply:
`thread_id` has a stream operator so it can be streamed to the log. So what you would be comparing is strings.

Reply:
So as I suspect, the thread id captured by `spdlog` is different from the serialized form of `std::this_thread::get_id()`. This is because spdlog is storing the hash of the `std::thread::id`: https://github.com/gabime/spdlog/blob/89ab1e679dc4446cce340230129c62092886ac81/include/spdlog/details/os-inl.h#L342

I could test for that, but that's an implementation detail I'd rather not hardcode. 

I could change it to explicitly capture the serialized form of `std::thread::id`, but I think this would just add overhead to logging because it requires using a `std::stringstream` for serialization, which has a lot more overhead than just appending strings. 

I don't think it really matters what the thread ids are so long as they are unique for different threads. However, that does mean I can't explicitly test for it in the unit tests, but I don't think that's a problem.

Reply:
As discussed offline, I now understand that the parsing of thread ID *is* exercised by the tests. Verifying the actual value is not important.